### ISSUE TITLE
Potential bug with wait()

### ISSUE THREAD

**Issue by ORESoftware** [SeleniumHQ/selenium#3780]
## Meta -
OS:   MacOS
Selenium Version:  

Browser:  Firefox

Browser Version:  Latest version, tmk

## Expected Behavior -

I am looking to upload a file to a Node.js server, using the following Node.js code:

```js
 before('upload file', {timeout: 30000}, h => {
      driver.get(cdthost + '/#!/home');
      let acq = acqs[0].companyName === 'DUMMY_ACQ' ? acqs[1] : acqs[0];
      let acqId = acq._id;
      driver.findElement(By.id(`open-editing-acqId-${acqId}`)).click();
      let file = path.resolve($root + '/csv-data/IT-DE-Jasper.csv');
      let el = driver.findElement(By.id(`import-file-acqId-${acqId}`));
      driver.wait(until.elementIsVisible(el), 1000);
      el.sendKeys(file);
      driver.findElement(By.id(`submit-file-acqId-${acqId}`)).click();
      return driver.wait(function () {
        return Promise.delay(100);
      }, 2000);
  });
```

What perplexes me is that when `driver.wait() `is called, the timeout will start running immediately. Instead of waiting for all the previous calls to resolve, it appears to start the timeout right away. 

Does that make sense?

In other words,

```js
  driver.doSomething1() // takes 5 seconds
  driver.doSomething2()  // takes 4 seconds
  driver.doSomething3() // takes 5 seconds
  return driver.wait(function(){
        return Promise.delay(100);
  }, 2000);
```

what seems to happen is that this will timeout because 2000 ms < is than 5+4+5 seconds.

Something is wrong, am I doing something wrong?

What I would expect is if y > x, then this will never timeout, am I right, or am I right?

```js
  return driver.wait(function(){
        return Promise.delay(x);
  }, y);

```

**Comment by jleyba** [comment]
`Promise.delay()` resolves to `undefined`, so your condition will never evaluate to true and will always time out. If you want a hard sleep/pause, use `driver.sleep(n)`.

**Comment by jleyba** [comment]
From the docs for [WebDriver#wait()](http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_IWebDriver.html#wait):

> For a Condition or function, the wait will repeatedly evaluate the condition until it returns a truthy value.
